src/main/webapp/src/app/admin-page/admin-page.component.ts

```diff
@@ -0,0 +1,341 @@
+      {
+        observe: 'response',
+        responseType: 'json',
+      }
+    );
+  }
+
+  /**
+   * Gets all plants this garden contains.
+   *
+   * Performs GET: /garden/{garden}/plant-list
+   *
+   * @return the http response.
+   */
+  getGardenPlantList(garden: string): Observable<HttpResponse<Array<String>>> {
+    return this.httpClient.get<Array<String>>(
+      '/garden/' + garden + '/plant-list',
+      {
+        observe: 'response',
+        responseType: 'json',
+      }
+    );
+  }
+
+  /**
+   * Gets user information for the specified user from
+   * the server. Returns an observable HTTP response.
+   *
+   * Performs GET: /user/{user}
+   *
+   * @param user The user reqested from the server.
+   * @return the http response.
+   */
+  getUserInfo(user: string): Observable<HttpResponse<User>> {
+    return this.httpClient.get<User>('/user/' + user, {
+      observe: 'response',
+      responseType: 'json',
+    });
+  }
+
+  /**
+   * Gets plant information for the specified plant from
+   * the server. Returns an observable HTTP response.
+   *
+   * Performs GET: /plant/{plant}
+   *
+   * @param plant The plant reqested from the server.
+   * @return the http response.
+   */
+  getPlantInfo(plant: string): Observable<HttpResponse<Plant>> {
+    return this.httpClient.get<Plant>('/plant/' + plant, {
+      observe: 'response',
+      responseType: 'json',
+    });
+  }
+
+  /**
+   * Creates garden summary.
+   *
+   * @param garden The garden to create an admin page of.
+   */
+  createGardenSummary(garden: string): void {
+    this.getGardenInfo(garden).subscribe({
+      next: (response: HttpResponse<Garden>) => {
+        // Successful responses are handled here.
+        this.gardenProfile = response.body;
+        this.isLoaded = true;
+        // Get rest of information
+        this.createGardenUserList(garden);
+        this.createGardenPlantList(garden);
+        this.gardenManager = 'Loading';
+        this.getUserInfo(this.gardenProfile.adminId).subscribe({
+          next: (response: HttpResponse<User>) => {
+            this.gardenManager = response.body.preferredName;
+          },
+          error: (error: HttpErrorResponse) => {
+            // Handle connection error
+            if (error.error instanceof ErrorEvent) {
+              console.error('Network error: ' + error.error.message);
+              this.gardenManager = 'Cannot fetch name';
+              return;
+            }
+            console.error('Unexpected error: ' + error.statusText);
+            this.gardenManager = 'Cannot fetch name';
+          },
+        });
+      },
+      error: (error: HttpErrorResponse) => {
+        // Handle connection error
+        if (error.error instanceof ErrorEvent) {
+          console.error('Network error: ' + error.error.message);
+          this.gardenProfile = null;
+          this.errorMessage = 'Cannot connect to GrowPod Server';
+          this.isLoaded = true;
+          return;
```

Comment:
nit: While it can be helpful to `return` to prevent additional nesting, since this is multiple conditional blocks it could be more readable to do a standard `if`, `elseif`, and `else` structure. Additionally instead of `return` with the first two sections, you can also just place lines  that are common to all blocks (setting `errorMessage` and `isLoaded`) before or after them.

Reply:
Will update. Thanks.